From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 19 Feb 2026 04:33:55 +0100
Subject: document-portal: Do not check the file-access for full permission
 IDs

In case a full permission ID (snap + snap app) is used, then snap
routine would fail, so prevent this by just getting the snap name

Origin: https://github.com/flatpak/xdg-desktop-portal/pull/1764
---
 document-portal/document-portal.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/document-portal/document-portal.c b/document-portal/document-portal.c
index a2c148d..1fa59e1 100644
--- a/document-portal/document-portal.c
+++ b/document-portal/document-portal.c
@@ -647,8 +647,13 @@ app_has_file_access (const char *target_permissions_id,
 
   if (g_str_has_prefix (target_permissions_id, "snap."))
     {
+      g_auto (GStrv) parts = NULL;
+
+      parts = g_strsplit (target_permissions_id + strlen ("snap."), "_", -1);
+
+      /* Snap does not support permission check for sub-applications */
       res = xdp_spawn (&error, "snap", "routine", "file-access",
-                        target_permissions_id + strlen ("snap."), path, NULL);
+                       parts[0], path, NULL);
 
       if (!res)
         {
